torch/utils/check_import.py

```diff
@@ -0,0 +1,146 @@
+# This script tries to figure out the reason of
+# `ImportError` on Windows.
+# Run it with `python check_import.py`.
+
+import ctypes
+import glob
+import os
+import sys
+import subprocess
+
+PY3 = sys.version_info >= (3, 0)
+
+TORCH_ROOT = os.path.dirname(os.path.dirname(__file__))
+PY_DLL_PATH = os.path.join(os.path.dirname(sys.executable), 'Library\\bin')
+TORCH_DLL_PATH = os.path.join(TORCH_ROOT, 'lib')
+NVTOOLEXT_HOME = os.getenv(
+    'NVTOOLSEXT_PATH', 'C:\\Program Files\\NVIDIA Corporation\\NvToolsExt')
+NV_ROOT = os.path.dirname(NVTOOLEXT_HOME)
+
+IS_CUDA = len(glob.glob(TORCH_ROOT + '\\_nvrtc*.pyd')) > 0
```

Comment:
Hmm... this seems like an awfully indirect way to detect if we're a CUDA build. Wouldn't `libcaffe2_gpu.dll` be a more direct way of testing this?